### PR TITLE
enhance(secret): use allow event system

### PR DESCRIPTION
**Pull request by ecrupper**
It will be helpful to share the same structure for allowed events for both repos and secrets.

### CODE REVIEW

File: library/actions/schedule.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package actions
+
+import "github.com/go-vela/types/constants"
+
+// Schedule is the library representation of the various actions associated
+// with the deploy event webhook from the SCM.
+type Schedule struct {
+	Run *bool `json:"created"`
```
Comment (wass3rw3rk) [inline library/actions/schedule.go:10]:
```suggestion
	Run *bool `json:"run"`
```

File: library/actions/schedule.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package actions
+
+import "github.com/go-vela/types/constants"
+
+// Schedule is the library representation of the various actions associated
+// with the deploy event webhook from the SCM.
```
Comment (wass3rw3rk) [inline library/actions/schedule.go:8]:
```suggestion
// with the schedule event webhook from the SCM.
```

**Review by wass3rw3rk — CHANGES_REQUESTED** [review]
first pass..

also, i didn't verify, but is checkEvents() actually unused as the linter claims? we can remove, if so.

**Comment by ecrupper** [comment]
> first pass..
> 
> also, i didn't verify, but is checkEvents() actually unused as the linter claims? we can remove, if so.

The linter really does have its moments

**Review by wass3rw3rk — DISMISSED** [review]

**Review by wass3rw3rk — APPROVED** [review]

**Pull request merged**